sklego/preprocessing/pandastransformers.py

```diff
@@ -34,7 +35,7 @@ def _nw_match_dtype(dtype, selection):
     raise ValueError(msg)
 
 
-def _nw_select_dtypes(df, include: str | list[str], exclude: str | list[str]):
+def _nw_select_dtypes(include: str | list[str], exclude: str | list[str], schema: dict[str, Any]):
```

Comment:
rather than passing `df` and repeatedly calling `df.schema`, we can just calculate `schema` once upfront and pass that